simulator/generation/plan.rs

```diff
@@ -251,31 +283,67 @@ impl Interaction {
             Self::Assertion(_) => {
                 unreachable!("unexpected: this function should only be called on queries")
             }
-            Interaction::Fault(_) => {
+            Self::Assumption(_) => {
```

Comment:
Btw you can just do sth like

```
_ => unreachable!("unexpected: ...")
```

to reduce some lines-of-code noise from the pattern matching. or alternatively

```
if let Self::Query(query) = self {
  ...
} else {
  unreachable!(...)
}
```

Reply:
That's a nice touch, adding it in the next commit.